ext/opcache/jit/zend_jit_trace.c

```diff
@@ -1587,6 +1587,23 @@ static zend_ssa *zend_jit_trace_build_tssa(zend_jit_trace_rec *trace_buffer, uin
 						}
 						ADD_OP2_TRACE_GUARD();
 					}
+					if (op1_type == IS_ARRAY
+					 && orig_op1_type != IS_UNKNOWN
```

Comment:
If op1_type is IS_ARRAY, how can orig_op1_type be IS_UNKNOWN (or IS_TRACE_REFERENCE for that matter)?

Reply:
> If op1_type is IS_ARRAY, how can orig_op1_type be IS_UNKNOWN (or IS_TRACE_REFERENCE for that matter)?

It cannot be IS_ARRAY here. Thanks. ``op1_type`` for references may be set to IS_ARRAY during code generation pass (after guard), but not during inference pass.